wgpu-hal/src/dx12/adapter.rs

```diff
@@ -311,6 +311,23 @@ impl super::Adapter {
         };
         features.set(wgt::Features::SHADER_INT64, int64_shader_ops_supported);
 
+        let atomic_int64_on_typed_resource_supported = {
+            let mut features9: crate::dx12::types::D3D12_FEATURE_DATA_D3D12_OPTIONS9 =
+                unsafe { mem::zeroed() };
+            let hr = unsafe {
+                device.CheckFeatureSupport(
+                    37, // D3D12_FEATURE_D3D12_OPTIONS9
+                    &mut features9 as *mut _ as *mut _,
+                    mem::size_of::<crate::dx12::types::D3D12_FEATURE_DATA_D3D12_OPTIONS9>() as _,
+                )
+            };
+            hr == 0 && features9.AtomicInt64OnTypedResourceSupported != 0
```

Comment:
I don't think this check is right.

Reply:
I'm asking for clarification on the features.

Reply:
To quote Jesse:

> - Raw buffers are required to be supported by SM6.6 without any additional optional caps.
> - A typed resource would be a RWBuffer<sometype> or RWTexture2D<sometype>
> - A descriptor heap resource is one accessed via ResourceDescriptorHeap[index]

So if we just want buffer and group shared (which is what webgpu supports) we want to check only `AtomicInt64OnGroupSharedSupported`